mesonbuild/mlog.py

```diff
@@ -119,9 +119,8 @@ def _log_error(severity, *args, **kwargs):
     else:
         assert False, 'Invalid severity ' + severity
 
-    if 'location' in kwargs:
-        location = kwargs['location']
-        del kwargs['location']
+    location = kwargs.pop('location', None)
+    if location:
```

Comment:
I think this is going to regress #3090.  I'm not sure what this change has to do with the rest of this PR.

Reply:
Not sure that fix is right, if it evaluate to false then False.subdir will raise exception the next line.

I do this because permittedkwargs can pass location=None.

Reply:
> Not sure that fix is right, if it evaluate to false then False.subdir will raise exception the next line.

No.   As I wrote there, if location is an ArgumentNode instance, it will evaluate as False when num_args is an empty sequence ( __len__() == 0, since it has no __bool__ method[1]), irrespective of the value of subdir.

[1] See https://docs.python.org/3/reference/datamodel.html

Perhaps the correct fix there is to explicitly add a __bool__ method to ArgumentNode which always returns True?